step-templates/sitecore-ship-publish.json

```diff
@@ -0,0 +1,114 @@
n=5)]\r\n    [string]$Username,\r\n\t[Parameter(Position=6)]\r\n    [string]$Password,\r\n\t[Parameter(Position=7)]\r\n    [string]$ListOfIDsInputPath\r\n)#>\r\n\r\n$SiteUrl = if($OctopusParameters[\"SiteUrl\"]){$OctopusParameters[\"SiteUrl\"]} else { throw \"SiteUrl required\"}\r\n$PublishTargets = if($OctopusParameters[\"PublishTargets\"]){$OctopusParameters[\"PublishTargets\"]} else { \"web\"}\r\n$PublishLanguaes = if($OctopusParameters[\"PublishLanguaes\"]){$OctopusParameters[\"PublishLanguaes\"]} else { \"en\"}\r\n$PublishMode = if($OctopusParameters[\"PublishMode\"]){$OctopusParameters[\"PublishMode\"]} else { \"smart\"}\r\n$ConnectionTimeOutInSeconds = if($OctopusParameters[\"ConnectionTimeOutInSeconds\"]){$OctopusParameters[\"ConnectionTimeOutInSeconds\"]} else { 300 }\r\n$Username = $OctopusParameters[\"Username\"]\r\n$Password = $OctopusParameters[\"Password\"]\r\n\r\n$previousDeployStep = $OctopusParameters[\"PreviousDeployStep\"];\r\n$ListOfIDsInputPath = ($OctopusParameters[\"Octopus.Action[\" + $previousDeployStep + \"].Output.Package.InstallationDirectoryPath\"] + \"_output.txt\")\r\n\r\n\r\nWrite-Host \"SiteUrl:\" $SiteUrl\r\nWrite-Host \"PreviousDeployStep:\" $previousDeployStep\r\nWrite-Host \"ListOfIDsInputPath:\" $ListOfIDsInputPath\r\nWrite-Host \"PublishTargets:\" $PublishTargets\r\nWrite-Host \"PublishLanguaes:\" $PublishLanguaes\r\nWrite-Host \"PublishMode:\" $PublishMode\r\nWrite-Host \"ConnectionTimeOutInSeconds:\" $ConnectionTimeOutInSeconds\r\nWrite-Host \"Username:\" $Username\r\nWrite-Host \"Password:\" $Password\r\n\r\n$publishUrl = \"$SiteUrl/services/publish/$PublishMode\"\r\n\r\n$targetDatabases = $PublishTargets -split \",\"\r\n$targetLanguages = $PublishLanguaes -split \",\"\r\n$pubishItems = @{\r\n\t\ttargets = $targetDatabases\r\n\t\tlanguages = $targetLanguages\r\n\t}\r\n\r\nif($PublishMode -eq \"listofitems\")\r\n{\r\n\tif (-not (Test-Path $ListOfIDsInputPath)) \r\n\t{\r\n\t\tthrow [System.IO.FileNotFoundException] \"listofitems publish: $ListOfIDsInputPath not found.\"\r\n\t}\r\n    [Collections.Generic.List[String]]$items = [IO.File]::ReadAllLines($ListOfIDsInputPath) \r\n\tWrite-Host $items -foregroundcolor cyan\r\n\t$pubishItems = @{\r\n\t\t\tTargetDatabases = $targetDatabases\r\n\t\t\tTargetLanguages = $targetLanguages\r\n\t\t\tItems = $items\r\n\t\t}\r\n}\r\n\t\r\nif($Username){\r\n\t$secpasswd = ConvertTo-SecureString $Password -AsPlainText -Force\r\n\t$mycreds = New-Object System.Management.Automation.PSCredential ($Username, $secpasswd)\r\n\tWrite-Host \"POST Publish $PublishMode with credentials to $publishUrl\" -foregroundcolor green\r\n\tInvoke-RestMethod -Uri $publishUrl -Method POST -ContentType \"application/json\" -Body ($pubishItems | ConvertTo-Json) -cred $mycreds\r\n}\r\nelse{\r\n\tWrite-Host \"POST Publish $PublishMode to $publishUrl\" -foregroundcolor green\r\n\tInvoke-RestMethod -Uri $publishUrl -Method POST -ContentType \"application/json\" -Body ($pubishItems | ConvertTo-Json)\r\n}",
```

Comment:
Misspelled parameter on this line `$PublishLanguaes = if($OctopusParameters["PublishLanguaes"]){$OctopusParameters["PublishLanguaes"]} else { "en"}`, should be `PublishLanguages`